pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -353,7 +353,8 @@ public synchronized ProducerStatsRecorderImpl getStats() {
             return null;
         }
         stats.reset();
-        producers.values().forEach(p -> stats.updateCumulativeStats(p.getStats()));
+        producers.forEach(
+                (partition, producer) -> stats.updateCumulativeStats(producer.getTopic(), producer.getStats()));
```

Comment:
`producer.getTopic()` -> `partition` ?

Reply:
here, we are keeping map of <topicName, stats> to maintain consistency with consumer-stats.
so, code change is correct where we are not using partition but we are using actual topic name of the partition.

Reply:
make sense